src/Umbraco.Web/Editors/ContentController.cs

```diff
@@ -388,48 +388,100 @@ public ContentItemDisplay GetEmptyByKey(Guid contentTypeKey, int parentId)
-            mapped.ContentApps = mapped.ContentApps.Where(x => x.Alias != "umbListView").ToList();
+            display.ContentApps = display.ContentApps.Where(x => x.Alias != "umbListView").ToList();
+            return display;
+        }
 
-            return mapped;
+        private ContentItemDisplay GetEmpty(IContentType contentType, int parentId)
+        {
+            var emptyContent = Services.ContentService.Create("", parentId, contentType, Security.GetUserId().ResultOr(0));
+            var mapped = MapToDisplay(emptyContent);
+            return CleanContentItemDisplay(mapped);
         }
 
         /// <summary>
-        /// Gets a collection of empty content items for all document types.
+        /// Gets an empty <see cref="ContentItemDisplay"/> for each content type in the IEnumerable, all with the same parent ID
         /// </summary>
-        /// <param name="contentTypeKeys"></param>
+        /// <remarks>Will attempt to re-use the same permissions for every content as long as the path and user are the same</remarks>
+        /// <param name="contentTypes"></param>
         /// <param name="parentId"></param>
-        [OutgoingEditorModelEvent]
-        public IDictionary<Guid, ContentItemDisplay> GetEmptyByKeys([FromUri] Guid[] contentTypeKeys, [FromUri] int parentId)
+        /// <returns></returns>
+        private IEnumerable<ContentItemDisplay> GetEmpties(IEnumerable<IContentType> contentTypes, int parentId)
         {
-            var result = new Dictionary<Guid, ContentItemDisplay>();
-
-            using var scope = _scopeProvider.CreateScope(autoComplete: true);
-            var contentTypes = Services.ContentTypeService.GetAll(contentTypeKeys).ToList();
+            var result = new List<ContentItemDisplay>();
+            var userId = Security.GetUserId().ResultOr(0);
+            var currentUser = Security.CurrentUser;
+            // We know that if the ID is less than 0 the parent is null.
+            // Since this is called with parent ID it's safe to assume that the parent is the same for all the content types.
+            var parent = parentId > 0 ? Services.ContentService.GetById(parentId) : null;
+            EntityPermissionSet permissions = null;
 
             foreach (var contentType in contentTypes)
             {
-                if (contentType is null)
+                var emptyContent = Services.ContentService.Create("", parentId, contentType, userId);
+
+                if (emptyContent is null)
                 {
                     throw new HttpResponseException(HttpStatusCode.NotFound);
                 }
 
-                result.Add(contentType.Key, GetEmpty(contentType, parentId));
+                // Get the path for the content, we get this for each just in case some content has Identity and a different path
+                // Since if that's the case we might not be able to re-use the permissions.
+                string path;
+                if (emptyContent.HasIdentity)
```

Comment:
Does the IContentService.Create method ever return a content with an identity? It's a factory method to create a brand new content item so not sure this will ever be the case?

Reply:
Ahh, that's very true, the Create method even calls `CreateContent` with `withIdentity` set to false, I can't believe I missed this 🙈